pkg/jsonfieldstest/jsonfieldstest.go

```diff
@@ -0,0 +1,39 @@
+package jsonfieldstest
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func jsonFieldsFor(cmd *cobra.Command) []string {
+	// This annotation is added by the `cmdutil.AddJSONFlags` function.
+	//
+	// This is an extremely janky way to get access to this information but due to the fact we pass
+	// around concrete cobra.Command structs, there's no viable way to have typed access to the fields.
+	//
+	// It's also kind of fragile because it's several hops away from the code that actually validates the usage
+	// of these flags, so it's possible for things to get out of sync.
+	stringFields, ok := cmd.Annotations["help:json-fields"]
+	if !ok {
+		return nil
+	}
+
+	return strings.Split(stringFields, ",")
+}
+
+type CommandConstructionFunc[T any] func(f *cmdutil.Factory, runF func(*T) error) *cobra.Command
```

Comment:
I'm not totally sure about this. The idea here is that it means we can do this:

```go
jsonfieldstest.ExpectCommandToSupportJSONFields(t, NewCmdView, []string{ ... })
```

Rather than this:

```go
jsonfieldstest.ExpectCommandToSupportJSONFields(t, NewCmdView(&cmdutil.Factory{}, nil), []string{ ... })
```

I think it's neat but it won't work if there are commands that actually try to call functions on the factory as part of construction. I don't recall any right now, and it's a very mechanical change to make if we discover there are, so I'm not worried about taking on a little risk right now.

Reply:
I think this makes sense!  I wouldn't mind a small comment on this type because it took me a hot minute to understand the approach here; basically the type represents the predominate run function signature we use in the various commands.  Not the end of the world if we don't but took a moment.

Reply:
Added a comment.